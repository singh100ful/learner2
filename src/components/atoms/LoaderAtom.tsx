import {colorPresets} from '@src/theme/colors';
import {verticalScale} from '@src/theme/scale';
import * as React from 'react';
import {ActivityIndicator, View} from 'react-native';

interface LoaderAtomProps {
  loading: boolean;
}

export const LoaderAtom: React.FC<LoaderAtomProps> = ({loading}) => {
  if (loading) {
    return (
      <View style={{paddingVertical: verticalScale(16)}}>
        <ActivityIndicator size={'large'} color={colorPresets.primaryCTA} />
      </View>
    );
  }
  return null;
};
