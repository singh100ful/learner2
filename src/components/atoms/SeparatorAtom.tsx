import {colorPresets} from '@src/theme/colors';
import {verticalScale} from '@src/theme/scale';
import * as React from 'react';
import {StyleSheet, Text, View} from 'react-native';

interface SeparatorAtomProps {}

export const SeparatorAtom: React.FC<SeparatorAtomProps> = ({}) => {
  return (
    <View style={styles.container}>
      <View style={styles.line} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    paddingVertical: verticalScale(10),
  },
  line: {flex: 1, backgroundColor: colorPresets.gray, height: 1},
});
