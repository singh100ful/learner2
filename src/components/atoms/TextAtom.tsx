import {defaultTexts} from '@src/theme/fonts';
import * as React from 'react';
import {Text, TextProps, TextStyle, View} from 'react-native';

interface TextAtomProps extends TextProps {
  text: string;
  style?: TextStyle;
  preset?: TextPresets;
}

export const TextAtom: React.FC<TextAtomProps> = ({
  text,
  style,
  preset = 'body',
  ...rest
}) => {
  const textStyle = [Presets[preset], style];
  return (
    <View>
      <Text
        {...rest}
        style={[{flexShrink: 1, flexWrap: 'wrap', color: '#000'}, textStyle]}>
        {text}
      </Text>
    </View>
  );
};

const Presets = {
  title: {...defaultTexts.title} as TextStyle,
  subTitle: {...defaultTexts.subTitle} as TextStyle,
  body: {...defaultTexts.body} as TextStyle,
  caption: {...defaultTexts.caption} as TextStyle,
};

export type TextPresets = keyof typeof Presets;
