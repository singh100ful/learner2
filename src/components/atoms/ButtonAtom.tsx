import * as React from 'react';
import {Pressable, PressableProps, ViewStyle} from 'react-native';
import {TextAtom} from './TextAtom';
import {colorPresets} from '@src/theme/colors';
import {moderateScale} from '@src/theme/scale';

interface ButtonAtomProps extends PressableProps {
  title: string;
  preset?: ButtonPresets;
}

export const ButtonAtom: React.FC<ButtonAtomProps> = ({
  title,
  preset = 'primary',
  ...rest
}) => {
  const viewStyle = Presets[preset];
  return (
    <Pressable
      {...rest}
      style={({pressed}) => [
        viewStyle,
        {
          opacity: pressed ? 0.7 : 1,
        },
      ]}>
      <TextAtom
        text={title}
        preset="subTitle"
        style={{color: colorPresets.white}}
      />
    </Pressable>
  );
};

const BASE_VIEW: ViewStyle = {
  backgroundColor: colorPresets.primaryCTA,
  padding: moderateScale(15),
  alignItems: 'center',
  justifyContent: 'center',
  borderRadius: moderateScale(8),
};

const SMALL_VIEW: ViewStyle = {
  backgroundColor: colorPresets.primaryCTA,
  padding: moderateScale(5),
  alignItems: 'center',
  justifyContent: 'center',
  maxWidth: moderateScale(120),
  borderRadius: moderateScale(5),
};

const Presets = {
  primary: {...BASE_VIEW} as ViewStyle,
  primaryDisabled: {
    ...BASE_VIEW,
    backgroundColor: colorPresets.gray,
  } as ViewStyle,
  small: {...SMALL_VIEW} as ViewStyle,
};

export type ButtonPresets = keyof typeof Presets;
