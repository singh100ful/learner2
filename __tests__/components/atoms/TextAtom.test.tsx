import {TextAtom} from '@src/components/atoms/TextAtom';
import * as React from 'react';
import renderer from 'react-test-renderer';

function sum(a: number, b: number) {
  return a + b;
}

test('sum numbers', () => {
  expect(sum(1, 2)).toEqual(3);
  expect(sum(2, 2)).toEqual(4);
});

describe('Text Component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<TextAtom text="Sample text" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with title preset', () => {
    const tree = renderer
      .create(<TextAtom text="Sample text" preset="title" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with subTitle preset', () => {
    const tree = renderer
      .create(<TextAtom text="Sample text" preset="subTitle" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with body preset', () => {
    const tree = renderer
      .create(<TextAtom text="Sample text" preset="body" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with caption preset', () => {
    const tree = renderer
      .create(<TextAtom text="Sample text" preset="caption" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
