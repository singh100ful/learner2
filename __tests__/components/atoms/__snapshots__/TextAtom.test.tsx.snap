// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component renders with body preset 1`] = `
<View>
  <Text
    style={
      [
        {
          "color": "#000",
          "flexShrink": 1,
          "flexWrap": "wrap",
        },
        [
          {
            "color": "#252A48",
            "fontSize": 24,
            "fontWeight": "400",
            "lineHeight": 28.56,
          },
          undefined,
        ],
      ]
    }
  >
    Sample text
  </Text>
</View>
`;

exports[`Text Component renders with caption preset 1`] = `
<View>
  <Text
    style={
      [
        {
          "color": "#000",
          "flexShrink": 1,
          "flexWrap": "wrap",
        },
        [
          {
            "color": "#252A48",
            "fontSize": 18,
            "fontWeight": "300",
            "lineHeight": 21.419999999999998,
          },
          undefined,
        ],
      ]
    }
  >
    Sample text
  </Text>
</View>
`;

exports[`Text Component renders with subTitle preset 1`] = `
<View>
  <Text
    style={
      [
        {
          "color": "#000",
          "flexShrink": 1,
          "flexWrap": "wrap",
        },
        [
          {
            "color": "#252A48",
            "fontSize": 24,
            "fontWeight": "700",
            "lineHeight": 28.56,
          },
          undefined,
        ],
      ]
    }
  >
    Sample text
  </Text>
</View>
`;

exports[`Text Component renders with title preset 1`] = `
<View>
  <Text
    style={
      [
        {
          "color": "#000",
          "flexShrink": 1,
          "flexWrap": "wrap",
        },
        [
          {
            "color": "#252A48",
            "fontSize": 30,
            "fontWeight": "700",
            "lineHeight": 35.699999999999996,
          },
          undefined,
        ],
      ]
    }
  >
    Sample text
  </Text>
</View>
`;

exports[`Text Component should render correctly 1`] = `
<View>
  <Text
    style={
      [
        {
          "color": "#000",
          "flexShrink": 1,
          "flexWrap": "wrap",
        },
        [
          {
            "color": "#252A48",
            "fontSize": 24,
            "fontWeight": "400",
            "lineHeight": 28.56,
          },
          undefined,
        ],
      ]
    }
  >
    Sample text
  </Text>
</View>
`;
